addons/common/XEH_postInit.sqf

```diff
@@ -263,6 +263,40 @@ if (_currentVersion != _previousVersion) then {
 
 call FUNC(checkFiles);
 
+// This handles clients joining that do not have ACE loaded
+if (isServer) then {
+    addMissionEventHandler ["PlayerConnected", {
+        params ["", "", "", "", "_owner"];
+
+        private _random = [];
+
+        // Make a random string of length 64
+        for "_i" from 0 to 31 do {
+            _random pushBack selectRandom GVAR(hexArray); // used GVAR(hexArray) because it was easy to use
+        };
+
+        _random = _random joinString "";
+
+        // Add the random suffix, so that it's much harder for the client to predict the function name and put preventive mesures in place
+        private _fncNameCheckAcePresence = format [QFUNC(checkAcePresence_%1), _random];
+        private _fncNameCheckErrorMessage = format [QFUNC(checkErrorMessage_%1), _random];
+
+        // Send functions to the connecting client
+        missionNamespace setVariable [_fncNameCheckAcePresence, /*compile toString */FUNC(checkAcePresence), _owner]; // compile toString removes isFinal status
+        missionNamespace setVariable [_fncNameCheckErrorMessage, /*compile toString */FUNC(checkErrorMessage), _owner];
+
+        // Wait for function to broadcast, then run function on client
+        [{
+            params ["_owner", "_fncNameCheckAcePresence", "_fncNameCheckErrorMessage"];
+
+            _fncNameCheckErrorMessage remoteExecCall [_fncNameCheckAcePresence, _owner];
+
+            // Delete the functions on the server
+            //missionNamespace setVariable [_fncNameCheckAcePresence, nil];
+            //missionNamespace setVariable [_fncNameCheckErrorMessage, nil];
```

Comment:
@LinkIsGrim I've completely reworked how the detection system works.

The idea is to make a system that is robust and cannot be hacked into.

What it does now is the server sends 2 functions, each with a unique name, to the connecting client and makes the client run `FUNC(checkAcePresence)`. `FUNC(checkAcePresence)` checks if ACE is properly installed (it can/should be expanded further to its current implementation). If ACE is detected, all is well, but if it isn't, then it runs `FUNC(checkErrorMessage)`. `FUNC(checkErrorMessage)` will spawn an error message that is very hard to remove without it quitting the game.

I've tested the system with ACE and ACE + TFAR on the host (because TFAR has an error message that makes the ACE one vanish, but if TFAR one is clicked away, it still quits the mission), client had no mods loaded.

However, there is a choice we have to make (if this system is adequate; I was thinking maybe we should implement a CBA setting to toggle this system): `FUNC(checkAcePresence)` and `FUNC(checkErrorMessage)` are both compiled with the `isFinal` flag being true, which means that any direct copy is also final. This means when we send the function over the network with `missionNamespace setVariable [_fncNameCheckAcePresence, FUNC(checkAcePresence), _owner];`, it will create a `missionNamespace` variable with the name `_fncNameCheckAcePresence`. This in turn means that every time someone connects, two new functions are created that cannot be deleted. I'm really not a fan of this, but currently I don't see any alternatives.

If a non-final version of the functions is sent, those copies can be deleted on the server. However, it also means that the client can overwrite them. There is a delay of about 1 second (in normal networking cases) on the client between receiving the functions and executing them, meaning there is ample time to scan all mission variables, finding the functions sent by the server and modifying them.

Any thoughts?

On a side note:
I'm getting
`10:46:27 Attempt to delete final function - ace_common_fnc_checkacepresence_953ef70819671adb14c96fce40c64a11f592a80bc5bb69378d5d5eeb1c520dfd` for an unknown reason, I will investigate this.